pyof/v0x01/common/phy_port.py

```diff
@@ -153,9 +153,9 @@ class PhyPort(GenericStruct):
     supported = UBInt32(enum_ref=PortFeatures)
     peer = UBInt32(enum_ref=PortFeatures)
 
-    def __init__(self, port_no=None, hw_addr=None, name=None, config=None,
-                 state=None, curr=None, advertised=None, supported=None,
-                 peer=None):
+    def __init__(self, port_no=None, hw_addr=None, name=None, config=0,
+                 state=PortState.OFPPS_STP_LISTEN, curr=0, advertised=0,
+                 supported=0, peer=0):
```

Comment:
They are not mandatory, but were chosen based on the OpenFlow specification.

`config` is a 'flags' field. The flags are defined in PortConfig, but if no flag is set (default for the OVS switch that generated the dump we have) the value is zero.

`state` is the STP state of the physical port - the default value is LISTEN.

The four features fields have zero as value when they are unavailable or unsupported by the switch - it happens to three of them in the OVS dump.